go/vt/vtgate/vindexes/cfc_test.go

```diff
@@ -0,0 +1,452 @@
+package vindexes
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/key"
+	"vitess.io/vitess/go/vt/proto/topodata"
+)
+
+func TestCFC(t *testing.T) {
+	t.Run("build cfc", func(t *testing.T) {
```

Comment:
This is a new pattern in this code base. I think I like it, but it made me scan for shared state between the tests where there shouldn't be. Don't mean that anything has to change. Just sharing my reaction :)